internal/pkg/template/templates/task/cf.yml

```diff
@@ -81,8 +77,9 @@ Resources:
       NetworkMode: awsvpc
       Cpu: !Ref TaskCPU
       Memory: !Ref TaskMemory
-      ExecutionRoleArn: !If [HasExecutionRole, !Ref ExecutionRole, !GetAtt DefaultExecutionRole.Arn]
+      ExecutionRoleArn: {{- if eq .ExecutionRole "" }} !GetAtt DefaultExecutionRole.Arn {{- else }} {{.ExecutionRole}} {{- end }}
```

Comment:
nit: we can remove the `HasExecutionRole` condition and the `ExecutionRole` parameter if they are no longer in use.